driver-core/src/main/java/com/datastax/driver/core/schemabuilder/KeyspaceOptions.java

```diff
@@ -0,0 +1,118 @@
+/*
+ *      Copyright (C) 2012-2015 DataStax Inc.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+package com.datastax.driver.core.schemabuilder;
+
+import com.google.common.base.Optional;
+
+import java.util.Map;
+
+/**
+ * The keyspace options used in CREATE KEYSPACE or ALTER KEYSPACE statements.
+ * <p/>
+ * Implementation notes: this class is abstract and not meant to use directly.
+ */
+public abstract class KeyspaceOptions<T extends KeyspaceOptions> extends SchemaStatement {
```

Comment:
I think we can simplify this a bit. This class could be public and concrete and the two subclasses declared in `CreateKeyspace` and `AlterKeyspace` could be deleted. To achieve this, we could modify the constructor to take the parent statement as an argument, stg along the lines of:

```java
public class KeyspaceOptions extends SchemaStatement {

    private final SchemaStatement statement;

    //...

    public KeyspaceOptions(SchemaStatement statement) {
        this.statement = statement;
    }

    //...

    @Override
    String buildInternal() {
        StringBuilder renderedStatement = new StringBuilder(statement.buildInternal());
        renderedStatement.append("\n\tWITH\n\t\t");
        //...
    }

}
```